mindee/parsing/common/inference.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict, Generic, List, Optional, TypeVar
+
+from mindee.parsing.common.page import TypePage
+from mindee.parsing.common.prediction import TypePrediction
+from mindee.parsing.common.product import Product
+from mindee.parsing.common.string_dict import StringDict
+
+
+class Inference(Generic[TypePrediction, TypePage]):
+    """Base Inference class for all predictions."""
+
+    product: Product
+    """Name and version of a given product, as sent back by the API."""
+    endpoint_name: Optional[str]
+    """Name of the endpoint for OTS APIs"""
+    endpoint_version: Optional[str]
+    """Version of the endpoint for OTS APIs"""
+    prediction: TypePrediction
+    """A document's top-level Prediction."""
+    pages: List[TypePage]
+    """A document's pages."""
+    is_rotation_applied: Optional[bool]
+    """Whether the document has had any rotation applied to it."""
+
+    def __init__(
+        self,
+        raw_prediction: StringDict,
+    ):
+        self.is_rotation_applied = None
+        if "is_rotation_applied" in raw_prediction:
+            self.is_rotation_applied = raw_prediction["is_rotation_applied"]
+        self.product = Product()
+        if "product" in raw_prediction and raw_prediction["product"]:
+            if "name" in raw_prediction["product"]:
+                self.product.name = raw_prediction["product"]["name"]
+            if "version" in raw_prediction["product"]:
+                self.product.version = raw_prediction["product"]["version"]
```

Comment:
Keep init logic within each class.

```suggestion
        self.product = Product(raw_prediction["product"])
```